libutils/buffer.c

```diff
@@ -111,7 +111,17 @@ int BufferCompare(const Buffer *buffer1, const Buffer *buffer2)
             /*
              * C String comparison
              */
-            return strcmp(buffer1->buffer, buffer2->buffer);
+            // Adding this as strcmp gives difference of the buffer values at aarch64 whereas it gives 1 or -1 in other platforms, in case compared char buffer were not same.
+            if (buffer1->buffer == buffer2->buffer)
+            {
```

Comment:
What is the need for this code here? Please justify.
`if (buffer1->buffer == buffer2->buffer) { ... }`
It seems this is useless here.